src/VernaContextProvider.tsx

```diff
@@ -1,36 +1,54 @@
 import * as React from 'react';
-import { PropsWithChildren, useState } from 'react';
+import { PropsWithChildren, useEffect, useState } from 'react';
 import { JSONSchema7 } from 'json-schema';
 import type { UiSchema, Widget } from '@rjsf/core';
 
 function functionNotSet() {
   throw new Error('function context not set');
 }
 
-export interface WidgetType {
+export interface WidgetsType {
   [name: string]: Widget;
 }
 
 interface VernaContextProps {
+  fullSchema: JSONSchema7;
+  setFullSchema: (newSchema: JSONSchema7) => void;
+  fullUiSchema: UiSchema;
+  setFullUiSchema: (newUiSchema: UiSchema) => void;
   schema: JSONSchema7;
   setSchema: (newSchema: JSONSchema7) => void;
+  getFormData: (localFormData: unknown) => unknown;
   uiSchema: UiSchema;
   setUiSchema: (newUiSchema: UiSchema) => void;
-  widgets: WidgetType;
-  setWidgets: (newWidgets: WidgetType) => void;
+  defaultValues: unknown;
+  setDefaultValues: (newDefaultValues: any) => void;
+  widgets: WidgetsType;
+  setWidgets: (newWidgets: WidgetsType) => void;
   readOnly: boolean;
   setReadOnly: (newMode: boolean) => void;
+  selector: string | undefined;
+  setSelector: (selector: string | undefined) => void;
 }
 
 const VernaContext = React.createContext<VernaContextProps>({
+  fullSchema: {},
+  setFullSchema: () => functionNotSet(),
+  fullUiSchema: {},
+  setFullUiSchema: () => functionNotSet(),
   schema: {},
   setSchema: () => functionNotSet(),
   uiSchema: {},
   setUiSchema: () => functionNotSet(),
+  getFormData: () => functionNotSet(),
+  defaultValues: {},
+  setDefaultValues: () => functionNotSet(),
   widgets: {},
   setWidgets: () => functionNotSet(),
   readOnly: true,
   setReadOnly: () => functionNotSet(),
+  selector: undefined,
+  setSelector: () => functionNotSet(),
```

Comment:
The list starts to be long. Please sort these lines alphabetically to improve readability.